File: backend-nestjs/src/modules/damage-reports/application/commands/update-damage-report-status/update-damage-report-status.command.ts
import { Command } from '@nestjs/cqrs';
import { DamageReport } from '../../../domain/damage-report';
import { type DamageReportStatus } from '../../../domain/damage-report.types';

export class UpdateDamageReportStatusCommand extends Command<DamageReport> {
  constructor(
    readonly id: string,
    readonly status: DamageReportStatus,
  ) {
    super();
  }
}

File: backend-springboot/src/main/java/com/greenalgeria/damagereport/application/command/UpdateDamageReportStatusCommand.java
package com.greenalgeria.damagereport.application.command;

import com.greenalgeria.damagereport.application.*;
import com.greenalgeria.damagereport.domain.DamageReportStatus;
import com.greenalgeria.shared.cqrs.Command;
import java.util.UUID;

public record UpdateDamageReportStatusCommand(UUID id, DamageReportStatus status)
        implements Command<DamageReportResponse> {}
